### PR TITLE
Add PIP22 regression tests

### PR DESCRIPTION
**Pull request by Andrew-Pohl**
Wasn't entirely sure how the scenario number was done so have left these blank for now.

### CODE REVIEW

**Comment by Andrew-Pohl** [comment]
@luyzdeleon FYI I can't add reviewers

**Review by luyzdeleon — CHANGES_REQUESTED** [review]
I believe before looking at the test themselves we should group scenarios in the following categories:

1. Before the upgrade height (the one that tells validators to be running on RC-0.9.0)
2. After the upgrade height.
2.a. After RSCAL is activated
2.b. After VEDIT is activated
3.c. When RSCAL and VEDIT are not active at the same time.

File: tests/PIP-22/behavioral-tests.feature [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    Then the stake of both node1 and node2 should have fallen by the same amount.
+    
+Scenario: xxx Make sure stake slashing does scale when RSCAL is enabled
+    Given that the user is running pocket version RC 0.9.0 and current block is greater than RSCAL activation
+    Given ServicerStakeFloorMultiplier = 15000000000, ServicerStakeWeightMultiplier = 1, ServicerStakeWeightCeiling = 60000000000 and ServicerStakeFloorMultiplierExponent = 1
+    And user is running 2 nodes 1 with 15001 (node1) pocket staked the other with 30001 (node2).
+    And each node is subjected to a single slashing event
+    Then the stake of node2 should have reduced twice as much than that of node1
+    
+Scenario: xxx Make sure stake can be updated friely within the same bin when VEDIT is disabled and RSCAL is diabled
```
Comment (luyzdeleon) [inline tests/PIP-22/behavioral-tests.feature:46]: Typo here: `friely`, should be `freely`

File: tests/PIP-22/behavioral-tests.feature [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    And wants to edit or modify his current stake within the network
+    When typing "pocket nodes stake <fromAddr> 15002000000 <chains> <serviceURI> <chainID> <fees>"
+    Then user should be prompted a success message/transaction:
+    | pocket nodes stake abf1df709a0cc486ac6db216ba9ed260e5597ba9 15002000000 0022 testnet 100000 |
+    And after a block have passed.
+    And user queries the node. 
+    | pocket query node <NodeAddress> | 
+    Then user finds that the stake amount has been updated.
+
+Scenario: xxx Make sure stake can be updated friely within the same bin when VEDIT is disabled and RSCAL is enabled
```
Comment (luyzdeleon) [inline tests/PIP-22/behavioral-tests.feature:58]: Typo here: `friely`, should be `freely`

File: tests/PIP-22/behavioral-tests.feature [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    And wants to edit or modify his current stake within the network
+    When typing "pocket nodes stake <fromAddr> 15002000000 <chains> <serviceURI> <chainID> <fees>"
+    Then user should be prompted a success message/transaction:
+    | pocket nodes stake abf1df709a0cc486ac6db216ba9ed260e5597ba9 15002000000 0022 testnet 100000 |
+    And after a block have passed.
+    And user queries the node. 
+    | pocket query node <NodeAddress> | 
+    Then user finds that the stake amount has been updated.
+
+Scenario: xxx Make sure stake can be updated friely within the same bin when VEDIT is enabled and RSCAL is disabled
```
Comment (luyzdeleon) [inline tests/PIP-22/behavioral-tests.feature:70]: Typo here: `friely`, should be `freely`

File: tests/PIP-22/behavioral-tests.feature [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    And wants to edit or modify his current stake within the network
+    When typing "pocket nodes stake <fromAddr> 15002000000 <chains> <serviceURI> <chainID> <fees>"
+    Then user should be prompted a success message/transaction:
+    | pocket nodes stake abf1df709a0cc486ac6db216ba9ed260e5597ba9 15002000000 0022 testnet 100000 |
+    And after a block have passed.
+    And user queries the node. 
+    | pocket query node <NodeAddress> | 
+    Then user finds that the stake amount has been updated.
+
+Scenario: xxx Make sure stake can not be updated friely within the same bin when VEDIT and RSCAL is enabled
```
Comment (luyzdeleon) [inline tests/PIP-22/behavioral-tests.feature:82]: Typo here: `friely`, should be `freely`

File: tests/PIP-22/behavioral-tests.feature [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    And wants to edit or modify his current stake within the network
+    When typing "pocket nodes stake <fromAddr> 15002000000 <chains> <serviceURI> <chainID> <fees>"
+    Then user should be prompted a success message/transaction:
+    | pocket nodes stake abf1df709a0cc486ac6db216ba9ed260e5597ba9 15002000000 0022 testnet 100000 |
+    And after a block have passed.
+    And user queries the node. 
+    | pocket query node <NodeAddress> | 
+    Then user finds that the stake amount has NOT been updated.
+
+Scenario: xxx Make sure stake can be updated friely to the next bin when VEDIT and RSCAL is enabled
```
Comment (luyzdeleon) [inline tests/PIP-22/behavioral-tests.feature:94]: Typo here: `friely`, should be `freely`

**Review by luyzdeleon — CHANGES_REQUESTED** [review]

File: tests/PIP-22/behavioral-tests.feature (hunk 1 above)
Comment (Andrew-Pohl) [reply tests/PIP-22/behavioral-tests.feature]: Whoops, copy and paste of a typo there. Fixed in [7b2ac1a](https://github.com/pokt-network/pocket-core-func-tests/pull/40/commits/7b2ac1abe8696c633960501368f60ee0fc244d0f)

**Review by luyzdeleon — APPROVED** [review]

**Review by emedrano2112 — COMMENTED** [review]
Proposed scenarios LGTM.

As mentioned in different threads, these are missing the Execution evidence as agreed on the contribution guide process, Success Criteria section. We agreed to proceed as is due to the tight timeline existing for this release, please provide when possible.

**Pull request merged**